qc_filtering/filter_functions.R

```diff
@@ -0,0 +1,124 @@
+## ---------------------------
+##
+## Script name: filter_functions.R
+##
+## Description:
+##
+## Args:
+##
+## Author: Georgia Doing
+##
+## Date Created: 2021-04-27
+##
+## Email: Georgia.Doing.GR@Dartmouth.edu
+##
+## ---------------------------
+##
+## Notes:
+##   
+##
+## ---------------------------
+library(RVenn)
+source('~/Dropbox (Hogan Lab)/Resources/Scripts/fsqn.R')
+
+hlog <- function(df, base=exp(1)){
+  # df: dataframe of gene expression to be logged
+  # base: base for log calculation
+  # returns: log of df with happy 0s for 0 or neg values
+  dfl <- data.frame(log(data.matrix(df),base))
+  dfl[dfl == -Inf] <- 0
+  dfl[dfl < 0] <- 0
+  dfl
+}
+
+MRnorm <- function(data){
+  # data: dataframe of gene expression counts to be normalized via DESeq2
+  meta <- data.frame(experiment = colnames(data))
+  dds <- DESeqDataSetFromMatrix(countData = 
+                                       ceiling(data), 
+                                     colData = meta, design = ~ experiment)
+  dds <- estimateSizeFactors(dds)
+  counts(dds, normalized=TRUE)
+}
+
+get_zeros <- function(data, lt= .1, ut = .9){
+  # data:
+  # lt:
+  # ut:
+  quantile(apply(
+    data, 2, 
+    FUN = function(x) sum( x  == 0)), probs = c(lt,ut))
+}
+
+get_hks <- function(data, hk_genes, lt=.2,ut=.98){
+  # data:
+  # hk_genes:
+  # lt: 
+  # ut:
+  quantile(apply(data[hks_genes,], 
+                 2, FUN = function(x) median(x)), probs = c(lt,ut))
+}
+
+filter_hks <- function(data, hk_genes=NA, hk_min=4, hk_max=6){
+  # data: dataframe of gene expression with rownames as gene names
+  # hk_genes: array of gene names
+  # hk_min: float of minimun mean expression, default 0
+  # hk_max: float of maximum mean expression, deafult infinity
+  # returns: array of bools for filtering columns of data
+  if(is.na(hk_genes)){
+    hk_genes <- sapply(c('ppiD','rpoD','rpoS','proC','recA','rpsL','rho','oprL','tipA','nadB','ampC'), function(x) name_to_PAO1(x))
```

Comment:
Maybe add a comment for where these hk genes are coming from

Reply:
Yep - I'll reference a paper for now here but should convert to link or PMID.